crates/evm/fuzz/src/strategies/invariants.rs

```diff
@@ -81,26 +87,32 @@ fn generate_call(
     let senders = Rc::new(senders);
     any::<prop::sample::Selector>()
         .prop_flat_map(move |selector| {
-            let (contract, func) = {
+            let (contract, func, abi) = {
                 let contracts = contracts.targets.lock();
                 let contracts =
                     contracts.iter().filter(|(_, (_, abi, _))| !abi.functions.is_empty());
                 let (&contract, (_, abi, functions)) = selector.select(contracts);
 
                 let func = select_random_function(abi, functions);
-                (contract, func)
+                (contract, func, abi.clone())
             };
 
             let senders = senders.clone();
             let fuzz_state = fuzz_state.clone();
             let fuzz_fixtures = fuzz_fixtures.clone();
             func.prop_flat_map(move |func| {
                 let sender = select_random_sender(&fuzz_state, senders.clone(), dictionary_weight);
-                let contract =
-                    fuzz_contract_with_calldata(&fuzz_state, &fuzz_fixtures, contract, func);
+                let contract = fuzz_contract_with_calldata(
+                    &fuzz_state,
+                    &fuzz_fixtures,
+                    contract,
+                    func,
+                    abi.clone(),
+                );
                 (sender, contract)
             })
         })
+        .prop_map(|(sender, call_details)| BasicTxDetails::new(sender, call_details))
```

Comment:
Please validate this addition, don't see any issue with but confirmation would be great

Reply:
yeah, it's just a code cleanup related change, that is to create a BasicTxDetails which is now a struct (used to be type)

Reply:
got it, seems reasonable to me but deferring to @klkvr since it's an implementation question :)